### ISSUE TITLE
Create config struct for hyperparameters

### ISSUE THREAD

**Issue by dorukkilitcioglu** [nickgreenquist/cu2rec#15]
Right now, we're passing a ton of hyperparameters back and forth. We should create a struct that carries all our hyperparameters, including learning rate, regularizing terms, etc.

Also modify the training loop so that we give the regularization param instead of the learning rate to the kernel.

**Comment by dorukkilitcioglu** [comment]
It might be better to put this into the constant memory of the GPU. Just asked Z about it, we'll see what he has to say.

**Comment by dorukkilitcioglu** [comment]
Struct is out, constant memory is in.

**Comment by nickgreenquist** [comment]
woot! go GPU memory constraints!

**Comment by dorukkilitcioglu** [comment]
Little update: apparently the constant memory accesses need to be in the same _file_ with the declaration. Otherwise, they point to _different_ locations (but don't cause any errors because CUDA). The options seem to be either to put everything that uses the hyperparams into the same file (training, loss, sgd), or do same `make` magic to tie all of them together at compile time. I'm leaning towards the second one, because it keeps the project more manageable.

**Comment by dorukkilitcioglu** [comment]
Merged in #19.
